appnav/src/main/kotlin/io/element/android/appnav/RootFlowNode.kt

```diff
@@ -88,11 +90,17 @@ class RootFlowNode @AssistedInject constructor(
     private fun observeLoggedInState() {
         authenticationService.isLoggedIn()
             .distinctUntilChanged()
+            .combine(
+                authenticationService.cacheIdx().onEach {
+                    Timber.v("cacheIdx=$it")
+                    matrixClientsHolder.removeAll()
+                }
+            ) { isLoggedIn, cacheIdx -> isLoggedIn to cacheIdx }
             .onEach { isLoggedIn ->
                 Timber.v("isLoggedIn=$isLoggedIn")
-                if (isLoggedIn) {
+                if (isLoggedIn.first) {
                     tryToRestoreLatestSession(
-                        onSuccess = { switchToLoggedInFlow(it) },
+                        onSuccess = { switchToLoggedInFlow(it, isLoggedIn.second) },
```

Comment:
It's unclear why you're passing the cache index here, it is not used in the `resolve()` method.

Reply:
If I do not pass the cacheIndex, Appyx will not reload the Node, because the state will be considered as unchanged. So it's like a cache buster.

Reply:
Then it is even more opaque than I thought :)
Maybe adding a comment that explains this side effect could help the future us?